### ISSUE TITLE
authorized_network doesn't work with shared vpc

### ISSUE THREAD

**Issue by vvinnich** [terraform-google-modules/terraform-google-memorystore#107]
### TL;DR

Couldn't create redis instance in project which use shared VPC network.
Works good for all other project with non shared vpc

### Observed behavior

│ Error: Error creating Instance: googleapi: Error 400: authorized_network must exist in the same project as redis instance
│ com.google.apps.framework.request.StatusException: <eye3 title='INVALID_ARGUMENT'/> generic::INVALID_ARGUMENT: authorized_network must exist in the same project as redis instance
│ Details:
│ [
│   {
│     "@type": "type.googleapis.com/google.rpc.BadRequest",
│     "fieldViolations": [
│       {
│         "description": "Invalid value: projects/infra/global/networks/dev-vpc",
│         "field": "instance.authorized_network"
│       }
│     ]
│   }
│ ]

### Terraform Configuration

```hcl
data "google_compute_network" "dev-network" {

  name = "dev-vpc"
  project = "infra"
}

resource "google_redis_instance" "default" {
  name           = "${var.env}-redis"
  project        = "dev-yt"
  tier           = var.tier
  memory_size_gb = var.memory_size_gb

  region = var.region

  authorized_network = data.google_compute_network.dev-network.id

  redis_version = "REDIS_6_X"

  maintenance_policy {
    weekly_maintenance_window {
      day = "DAY_OF_WEEK_UNSPECIFIED"
      start_time {
        hours   = 0
        minutes = 30
        seconds = 0
        nanos   = 0
      }
    }
  }
}
```

### Terraform Version

```sh
1.2.3
```

**Comment by tknodell-recurly** [comment]
Try setting
```  connect_mode            = "PRIVATE_SERVICE_ACCESS"```

ref - https://cloud.google.com/memorystore/docs/redis/troubleshoot-issues#incompatible_use_of_direct_peering_connection_mode_and_shared_vpc_network_during_instance_creation

**Comment by vvinnich** [comment]
@tknodell-recurly  it works. thx a lot
